cdap-app-templates/cdap-program-report/src/main/java/co/cask/cdap/report/proto/Sort.java

```diff
@@ -0,0 +1,75 @@
+/*
+ * Copyright © 2018 Cask Data, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+package co.cask.cdap.report.proto;
+
+import co.cask.cdap.report.util.ReportField;
+
+import java.util.ArrayList;
+import java.util.List;
+
+/**
+ * A class represents the field to sort the report by and the order of sorting by this field.
+ */
+public class Sort extends ReportGenerationRequest.Field {
+  private final Order order;
+
+  public Sort(String fieldName, Order order) {
+    super(fieldName);
+    this.order = order;
+  }
+
+  /**
+   * @return the sorting order with this field
+   */
+  public Order getOrder() {
+    return order;
+  }
+
+  @Override
+  public List<String> getErrors() {
+    ArrayList<String> errors = new ArrayList<>();
+    errors.addAll(super.getErrors());
+    ReportField sortField = ReportField.valueOfFieldName(getFieldName());
+    if (sortField == null) {
```

Comment:
this check and `super.getErrors()` are the same, get rid of the super getErrors call